docs/architecture/MLOps_process.md

```diff
@@ -0,0 +1,207 @@
+
+### Manage artifacts with Azure Machine Learning ###
+
+Azure Machine Learning is a service that helps orchestrate the development and management of machine learning artifacts. The following are the most important components for pipeline implementations:
+
+* **Azure ML SDK for Python**: used to create all Azure ML components and communicate to Azure ML service.
+* **Azure ML Compute**: allows us to create and manage clusters to run pipelines. It can be just one cluster for all steps or different clusters per step. Storage can be mounted to compute cluster instances.
+* **Azure ML Model Repository**: used to register and manage models, as well as track model versions. A model is a single file or a folder containing a set of files.
+* **ML Pipeline SDK (as a part of Azure ML SDK)**: used to define training pipelines including steps and their configurations, compute cluster parameters, sequences.
+* **ParallelRunStep and PythonScriptStep**: commonly used classes from Azure ML SDK in pipelines to run code in parallel or on a single instance. Other step types include MPI training or training on Databricks.
+
+Let’s show how the above example looks like from technologies perspective:
+
+![Pipeline in Azure ML](../media/mlops_arch_pipe_aml.png)
+
+### ML Pipelnes and data scientists ###
+
+Data scientists often prefer to use Jupyter Notebooks to do their experiments. Asking them to implement Azure ML pipelines from the beginning could hinder their productivity and creativity for the following reasons:
+
+* Not every Jupyter notebook is going to be a pipeline in production. In some cases data scientists need to test many different approaches and ideas before pick the right one.
+* It’s too hard to design a pipeline and all the steps at the beginning of the project because you don’t know the most critical blocks and the right split into steps.
+
+Initial experiments can be done in a notebook, however, there are several signs that indicate maybe it's time to move experiments from a notebook to Azure ML Service:
+
+* The code and process are stable, and we can move it from the research to development phase
+* Data scientists would like to run the same experiments with different parameters at the same time
+* Data Scientists need additional compute resources because training on a single VM or a local computer takes too long
+
+Once you see any of these signs, it’s time to wrap experiments using Azure ML Pipeline SDK. We would recommend using the three-step approach:
+
+* **Step 1. Refactor the notebook to Python files.** The primary goal of this step is to move all methods/classes to separate Python files and make them independent from the execution environment.
```

Comment:
I might add something like "Focus on pulling out and generalizing components which you expect to be reused across multiple experiments, allowing future notebooks to be implemented more easily using the refactored Python code."

Reply:
Agreed and updated.